src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -62,11 +70,38 @@ public function process(ContainerBuilder $container)
                 $tmpContainer->addObjectResource($configuration);
             }
 
-            foreach ($exprLangProviders as $provider) {
-                $tmpContainer->addExpressionLanguageProvider($provider);
+            $resetPlaceholders = false;
+            if (null !== $envTypes && method_exists(BaseNode::class, 'setPlaceholder')) {
+                $defaultEnvBag = new ParameterBag($parameters);
+                foreach ($resolvingBag->getEnvPlaceholders() as $env => $placeholders) {
+                    $prefix = (false === $i = strpos($env, ':')) ? 'string' : substr($env, 0, $i);
+                    $types = isset($envTypes[$prefix]) ? $envTypes[$prefix] : array('string');
+                    $default = ($hasEnv = (false === $i && $defaultEnvBag->has("env($env)"))) ? $defaultEnvBag->get("env($env)") : null;
+                    if (null !== $default && !in_array($type = gettype($default), $types, true)) {
+                        throw new LogicException(sprintf('Invalid type for env parameter "env(%s)", expected "%s" got "%s".', $env, implode('", "', $types), $type));
+                    }
+                    $values = array();
+                    foreach ($types as $type) {
+                        $values[$type] = $hasEnv ? $default : (isset(self::$typeFixtures[$type]) ? self::$typeFixtures[$type] : null);
+                    }
+                    foreach ($placeholders as $placeholder) {
+                        $resetPlaceholders = true;
+                        BaseNode::setPlaceholder($placeholder, $values);
+                    }
+                }
             }
 
-            $extension->load($config, $tmpContainer);
+            try {
+                foreach ($exprLangProviders as $provider) {
+                    $tmpContainer->addExpressionLanguageProvider($provider);
+                }
+
+                $extension->load($config, $tmpContainer);
+            } finally {
+                if ($resetPlaceholders) {
```

Comment:
I would remove this "if" and unconditionally call resetPlaceholders, the end result will be the same

Reply:
the condition would be `method_exists` check then =/

Reply:
OH OK, I missed the method_exists check L74, fine as is then!